src/gui/uithememanager.cpp

```diff
@@ -58,6 +61,98 @@ namespace
 
         return {};
     }
+
+    QByteArray readFile(const QString &fileName)
+    {
+        QFile file(fileName);
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+        {
+            LogMsg(UIThemeManager::tr("UITheme - Failed to open \"%1\". Reason: %2")
+                    .arg(QFileInfo(fileName).fileName(), file.errorString())
+                   , Log::WARNING);
+            return {};
+        }
+
+        return file.readAll();
+    }
+
+    class QRCThemeSource : public UIThemeSource
+    {
+    public:
+        QByteArray styleSheet() override
+        {
+            return readFile(QRC_THEME_DIR + STYLESHEET_FILE_NAME);
+        }
+
+        QByteArray config() override
+        {
+            return readFile(QRC_THEME_DIR + CONFIG_FILE_NAME);
+        }
+
+        QString iconPath(const QString &iconId) override
+        {
+            return findIcon(iconId, QRC_ICONS_DIR);
+        }
+
+    private:
+        /* static */ const QString QRC_THEME_DIR {":/uitheme/"};
```

Comment:
Why are these constants declared here and not in the same place as the others?

Reply:
they are not needed in the global scope, only to the concerned class.

Reply:
> they are not needed in the global scope, only to the concerned class.

`STYLESHEET_ICONS_DIR` too.

Reply:
I would throw away `QRC_ICONS_DIR` (just use `QRC_THEME_DIR + THEME_ICONS_DIR` like in other similar cases) and make `QRC_THEME_DIR` global to avoid declaring the similar constant `STYLESHEET_ICONS_DIR`.

Reply:
STYLESHEET_ICONS_DIR is global info, while QRC_* is local, also STYLESHEET_ICONS_DIR and QRC_THEME_DIR should start with same pointer AFAIK

Reply:
`STYLESHEET_ICONS_DIR` it does not point to the icon folder. Rather, it is some placeholder that replaces the name of the theme folder.

Reply:
> STYLESHEET_ICONS_DIR and QRC_THEME_DIR should start with same pointer

Sorry , I don't understand what do you mean...

Reply:
And Yes, these constants should be static, if they are the same for each class instance.

Reply:
```suggestion
        static const QString QRC_THEME_DIR {QStringLiteral(":/uitheme/")};
```

Reply:
> STYLESHEET_ICONS_DIR it does not point to the icon folder

Its icons dir used by stylesheet.qss. it's as the name suggests stylesheet icons dir or stylesheet resource dir.

> And Yes, these constants should be static, if they are the same for each class instance.

there will only be one instance of theme source, and this class is instantiated conditionally, making these variables static doesn't add much value but increases memory consumption (it won't be much but ....)

Reply:
> > STYLESHEET_ICONS_DIR it does not point to the icon folder
> 
> Its icons dir used by stylesheet.qss. it's as the name suggests stylesheet icons dir or stylesheet resource dir.

STYLESHEET_ICONS_DIR is confusing. Maybe STYLESHEET_RESOURCES_DIR would be better?..

Reply:
> > And Yes, these constants should be static, if they are the same for each class instance.
> 
> there will only be one instance of theme source

The class itself knows nothing about its usage.

> making these variables static doesn't add much value but increases memory consumption

Well, at least delete these comments so as not to confuse anyone.